lib/icinga/downtime.cpp

```diff
@@ -369,19 +369,19 @@ void Downtime::RemoveDowntime(const String& id, bool includeChildren, bool cance
 
 	String config_owner = downtime->GetConfigOwner();
 
-	if (!config_owner.IsEmpty() && !expired) {
+	if (!config_owner.IsEmpty() && removalReason != DowntimeExpired && removalReason != DowntimeRemovedByConfigOwner) {
```

Comment:
Now as you have only 3 enum values, you can just change != to == in f9ae1d58cbab7fd2432a90643c73a0368f7b8495. I.e.:

```suggestion
	if (!config_owner.IsEmpty() && removalReason == DowntimeRemovedByUser) {
```